src/api/bridge.js

```diff
@@ -65,9 +65,31 @@ export const maxTotalBalance = async ({decimals, symbol, contract}) => {
   }
 };
 
-export const initiateWithdraw = async ({recipient, amount, decimals, contract}) => {
-  return sendL2Transaction(contract, 'initiate_withdraw', {
-    l1Recipient: parseToFelt(recipient),
-    amount: parseToUint256(amount, decimals)
-  });
+const initiateWithdrawCall = ({bridge, recipient, amount, decimals}) => {
+  return {
+    contract: bridge,
+    method: 'initiate_withdraw',
+    args: {
+      l1Recipient: parseToFelt(recipient),
+      amount: parseToUint256(amount, decimals)
+    }
+  };
+};
+
+const increaseAllowanceCall = ({bridge, token, amount, decimals}) => {
+  return {
+    contract: token,
+    method: 'increaseAllowance',
+    args: {
+      spender: bridge.address,
+      amount: parseToUint256(amount, decimals)
```

Comment:
Do we want to increase the allowance only for the amount of the tx? this means we will do it for every tx. why not increase it once with a higher value (2^249 for example)?

Reply:
@bbrandtom WDYT?

Reply:
Right! missed that you're using multicall, cool!